### PR TITLE
[Cu-357bfm4] Get User Information

### PR DESCRIPTION
**Pull request by TaninDean**

### CODE REVIEW

**Comment by sirateek** [comment]
Task linked: [CU-357bfm4 Get user information API](https://app.clickup.com/t/357bfm4)

File: service/getuserinfo.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	CheckGetUserInformation(citizenID string) (model.Population, bool)
+}
+
+func NewGetUserInformtaionService(getUserInfoRepository repository.GetUserInformationRepository) GetUserInformationService {
+	return &getUserInformationService{
+		getUserInfoRepository: getUserInfoRepository,
+	}
+}
+
+func (g *getUserInformationService) CheckGetUserInformation(citizenID string) (model.Population, bool) {
```
Comment (sirateek) [inline service/getuserinfo.go:23]: Why return `bool` ?

File: main.go [hunk 2]
```diff
@@ -33,18 +33,14 @@ func main() {
...
+	getUserInfomationService := service.NewGetUserInformtaionService(getUserInformationRepository)
 
-	// New Handler
-	validityHandler := handler.NewValidityHandler(jwtService, validityService)
+	getUserInformationHandler := handler.NewGetUserInformationHandler(getUserInfomationService)
 
-	// Init Gin.
 	server := httpserver.NewHttpServer()
-	server.GET("/validity", handler.AuthorizeJWT(jwtService, appConfig), validityHandler.Validity)
+	server.GET("/user/info", handler.AuthorizeJWT(jwtService, appConfig), getUserInformationHandler.GetuserInfo)
```
[the remark this answers is not in the document]
Comment (TaninDean) [reply main.go:43]: i think i happen when merge your branch but i select cuurent. i will fix it as soon as possible

File: service/getuserinfo.go (hunk 1 above)
Comment (TaninDean) [reply service/getuserinfo.go:23]: I will make it for checking query citizenID is success or not. If success I will return info of user but if not I will return BadRequest

Comment (sirateek) [reply service/getuserinfo.go:23]: You should return `error` out instead to display the error precisely.

Also in handler, you should check if err returned from service is nil or not.

Comment (TaninDean) [reply service/getuserinfo.go:23]: Oh. I see how it is.I will change it to check error instead of boolean.

File: handler/userinfo.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package handler
```
Comment (sirateek) [inline handler/userinfo.go:1]: Please also change the file name to user.go krub

File: repository/getuserinfo.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package repository
```
Comment (sirateek) [inline repository/getuserinfo.go:1]: Please also change the file name to population.go krub

File: service/getuserinfo.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package service
+
```
Comment (sirateek) [inline service/getuserinfo.go:2]: Please also change the file name to `user.go` krub.

File: repository/getuserinfo.go [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (nabhan-au) [inline repository/getuserinfo.go:2]: Did you renamed wrong file?

Comment (TaninDean) [reply repository/getuserinfo.go:2]: Oh!. My bad i will fix it now

File: service/user.go [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+	CheckGetUserInformation(citizenID string) (model.Population, error)
+}
+
+func NewUserService(populationRepository repository.PopulationRepository) UserService {
+	return &userService{
+		populationRepository: populationRepository,
+	}
+}
+
+func (g *userService) CheckGetUserInformation(citizenID string) (model.Population, error) {
```
Comment (nabhan-au) [inline service/user.go:23]: Why we need to named it as CheckGet

Comment (TaninDean) [reply service/user.go:23]: it have case that citizenid doesn't match so it will check when query  so i name it as Check

Comment (nabhan-au) [reply service/user.go:23]: For this case I think `GetUserInformation` is better that `CheckGetUserInformation`

Comment (TaninDean) [reply service/user.go:23]: Ok .I will change them immidiately

File: handler/user.go [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+func (g *userHandler) GetuserInfo(gi *gin.Context) {
+	userInfo, err := g.service.CheckGetUserInformation(gi.Param("CitizenID"))
+	if err == nil {
+		gi.JSON(http.StatusOK, gin.H{
+			"info": userInfo,
+		})
+	} else {
+		gi.Status(http.StatusBadRequest)
```
Comment (nabhan-au) [inline handler/user.go:27]: I think it should be error 500 because err doesn't return only in citizenID not found case

Comment (nabhan-au) [reply handler/user.go:27]: You may handle it separate case

Comment (TaninDean) [reply handler/user.go:27]: I see. i will work on it

**Review by nabhan-au — APPROVED** [review]

**Review by sirateek — APPROVED** [review]

**Pull request merged**